execution/executor/src/db_bootstrapper.rs

```diff
@@ -24,19 +24,38 @@ use move_core_types::move_resource::MoveResource;
 use std::collections::btree_map::BTreeMap;
 use storage_interface::{state_view::VerifiedStateView, DbReaderWriter, TreeState};
 
-pub fn bootstrap_db_if_empty<V: VMExecutor>(
+pub fn generate_waypoint<V: VMExecutor>(
     db: &DbReaderWriter,
     genesis_txn: &Transaction,
-) -> Result<Option<Waypoint>> {
+) -> Result<Waypoint> {
     let tree_state = db.reader.get_latest_tree_state()?;
-    if !tree_state.is_empty() {
-        return Ok(None);
+
+    let committer = calculate_genesis::<V>(db, tree_state, genesis_txn)?;
+    Ok(committer.waypoint)
+}
+
+pub fn bootstrap_genesis<V: VMExecutor>(
```

Comment:
I'd call it `maybe_bootstrap_x()` or `apply_genesis_if_not_yet`, etc. And document the return value.